### PR TITLE
WIP File Import

### PR DESCRIPTION
**Pull request by buob**

### CODE REVIEW

File: core/src/main/web/app/mainapp/mainapp.js [hunk 1]
```diff
@@ -355,6 +356,16 @@
...
             },
+            fromImport: function(sessionId) {
+              var notebook = $sessionStorage.importedNotebook;
+              var notebookUri = null;
+              var uriType = null;
+              var readOnly = true;
+              var format = null;
+              var notebookModel = bkNotebookVersionManager.open(notebook);
+              loadNotebookModelAndResetSession(
+                  notebookUri, uriType, readOnly, format, notebookModel, false, sessionId, false);
```
Comment (buob) [inline core/src/main/web/app/mainapp/mainapp.js:369]: I'm just loading the notebook here right from the JSON received from the server which is resulting in inputs as arrays, yielding empty input fields.

**Comment by scottdraves** [comment]
The problem comes from not using the standard importer like this: https://github.com/twosigma/beaker-notebook/blob/master/core/src/main/web/app/mainapp/mainapp.js#L328

looks like that wants the document as a string instead of already parsed into json so maybe need to change the servlet to return text/plain?

also, i am concerned by the use of ngstorage.  i don't know what this is and why it's needed, the other import and file IO functions pass the data through callbacks instead of storing it in this relatively global location.

**Comment by buob** [comment]
Yeah, I don't love it either.  Any thoughts on a better technique given our strategy of using the redirect in order to load a notebook?

**Comment by scottdraves** [comment]
You could skip doing the redirect and instead directly connect the menu item to openNotebook with a newly defined uriType "fileupload" that opens the native dialog with a callback that has the text.

**Comment by scottdraves** [comment]
but we can always leave that as an improvement for the future, for now it's fine to just get it working by getting the document as a string instead of already parsed as json.

**Pull request merged**